File: Template/Math/fastpow.cpp
long long pow(long long base, long long exp)
{
	long long result = 1;
	while(exp)
	{
		if(exp&1) result *= base;
		base *= base;
		exp >>= 1;
	}
	return result;
}

long long powWithMod(long long base, long long exp, long long mod)
{
	long long result = 1;
	while(exp)
	{
		if(exp&1) result = (result*base) % mod;
		base = (base*base) % mod;
		exp >>= 1;
	}
	return result;
}

File: Template/Python/FastPow.py
# https://www.acmicpc.net/problem/1629
import sys

def powWithMode(base, exp, mod):
  ret = 1
  while exp:
    if exp & 1:
      ret = (ret * base) % mod
    base = (base * base) % mod
    exp >>= 1

  return ret

a,b,c = map(int, sys.stdin.readline().split())

print(powWithMode(a, b, c))